app/assets/javascripts/member-facing/redirectors.js

```diff
@@ -0,0 +1,43 @@
+const RegisterMemberRedirector = {
+  attemptRedirect:  function(followUpUrl, member) {
+
+    if(typeof(member) !== 'object') {
+      member = window.champaign.personalization.member;
+    }
+
+    if ( shouldMemberRegister() ) {
+      redirectToRegistration();
+      return true;
+    } else {
+      return false;
+    }
+
+    function shouldMemberRegister() {
+      return !member.registered;
+    }
+
+    function redirectToRegistration() {
```

Comment:
@rodrei ok, my bad. I wasn't aware of that conversation. I'd just think that the es6 syntax for function and method expressions is so much better. Here's one example where I'd use the arrow syntax.